### PR TITLE
Optimize interpreter: Move evaluation of expressions and statements to separate files

### PR DESCRIPTION
**Pull request by turbolent**
⚠️ Depends on #681
Work towards #678

The code for the interpreter is all in one file. Split it into separate files to make it more maintainable.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/interpreter/interpreter_expression.go [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+		right, ok := right.(EquatableValue)
+		if !ok {
+			return false
+		}
+		return left.Equal(interpreter, right)
+
+	case *ArrayValue,
+		*DictionaryValue:
+		// TODO:
+		return false
```
Comment (SupunS) [inline runtime/interpreter/interpreter_expression.go:259]: we might not need this, if `ArrayValue` and `DictionaryValue` implement the  `EquatableValue`, when we add it?

File: runtime/interpreter/interpreter_expression.go [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+
+		case *SomeValue:
+			result = typedResult.Value
+
+		default:
+			panic(errors.NewUnreachableError())
+		}
+	}
+
+	value := result.(Value)
```
Comment (SupunS) [inline runtime/interpreter/interpreter_expression.go:387]: can omit the cast. `result` is already a `Value`

File: runtime/interpreter/interpreter_statement.go [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+	rightValue := rightGetterSetter.get()
+	if interpreter.movingStorageIndexExpression(swap.Right) != nil {
+		rightGetterSetter.set(NilValue{})
+	}
+
+	// Add right value to left target
+	// and left value to right target
+
+	rightValueCopy := interpreter.copyAndConvert(rightValue.(Value), rightType, leftType)
+	leftValueCopy := interpreter.copyAndConvert(leftValue.(Value), leftType, rightType)
```
Comment (SupunS) [inline runtime/interpreter/interpreter_statement.go:384]: Could do the same.
```suggestion
	rightValueCopy := interpreter.copyAndConvert(rightValue, rightType, leftType)
	leftValueCopy := interpreter.copyAndConvert(leftValue, leftType, rightType)
```

**Review by SupunS — APPROVED** [review]
This is great! Makes it so much easier to read the codebase.

I guess this is simply a 'move', with no new changes?

**Comment by turbolent** [comment]
Yep, no changes, just a move 👍

File: runtime/interpreter/interpreter_expression.go (hunk 1 above)
Comment (turbolent) [reply runtime/interpreter/interpreter_expression.go:259]: Right, the TODO is for making arrays and dictionaries equatable (this also means allowing this statically)

File: runtime/interpreter/interpreter_expression.go (hunk 2 above)
Comment (turbolent) [reply runtime/interpreter/interpreter_expression.go:387]: Nice!

Comment (turbolent) [reply runtime/interpreter/interpreter_expression.go]: Cleaned up in 9663ee9947b549463d4817a888cfa595c8180b63

File: runtime/interpreter/interpreter_statement.go (hunk 3 above)
Comment (turbolent) [reply runtime/interpreter/interpreter_statement.go]: Good catch! 👍 

Also removed the other unnecessary casts to `Value` in 9663ee9947b549463d4817a888cfa595c8180b63

**Pull request merged**